src/format/scan.rs

```diff
@@ -238,6 +296,16 @@ where
     };
     s = &s[1..];
 
+    // special check for `Z` to return more accurate error `INVALID`.
```

Comment:
This should be a separate commit.

Reply:
I decided to drop this. After adding more test cases around this, I realized it was somewhat tunnel-vision change (it was not as helpful as I thought).  The benefit is middling in exchange for PR feature creep.